### ISSUE TITLE
 Resolve CodeQL query alert 57

### ISSUE THREAD

**Issue by roslynwythe** [hackforla/website#6479]
### Prerequisite
1. Be a member of Hack for LA. (There are no fees to join.) If you have not joined yet, please follow the steps on our [Getting Started page](https://www.hackforla.org/getting-started).
2. Before you claim or start working on an issue, please make sure you have read our [How to Contribute to Hack for LA Guide](https://github.com/hackforla/website/blob/7f0c132c96f71230b8935759e1f8711ccb340c0f/CONTRIBUTING.md).

### Overview
We need to analyze the query "Use of returnless function" which appears in the CodeQL alert [57](https://github.com/hackforla/website/security/code-scanning/57) then either recommend dismissal of the alert or update code to resolve the alert. 

### Action Items
- [x] DO NOT DISMISS ANY ALERTS.  Dismissal of alerts should be done by dev leads only after review of the recommendation
- [x] Browse to the links in the following two Action Item and read the contents.  Click "See More" to view Recommendations, Examples and References.  
- [x] https://github.com/hackforla/website/security/code-scanning/57
- [x] In a comment in this issue, add your analysis and recommendations.  For each alert, the recommendation can be one of the following: `dismiss as test`, `dismiss as false positive`, `dismiss as won't fix`, or `update code`. 
- [x] If the recommendation is to update code:
   - [x] create an issue branch and proceed with the code update
   - [x] Use docker to test locally, ensuring that there are no changes to any affected webpage(s)
   - [x] proceed with pull request in the usual manner 
- [ ] If the recommendation is to dismiss, describe your reason for dismissal in the comment, then move the issue to `Questions/In Review` and apply the label `ready for dev lead`.   

### For merge team/dev lead
- [ ] If recommendation to dismiss is approved, dismiss the alert with a comment, then close the issue as completed.
- [ ] When this issue is closed please check it off (under "Issues") in #5159.  If all child issues are closed, close #5159 as completed.  
  
### Resources/Instructions
- [GitHub CodeQL documentation](https://codeql.github.com/docs/codeql-overview/about-codeql/)
- This issue is part of #5060

**Comment by aqandrew** [comment]
Availability:
  - Wednesday 3/27 10am-3pm
  - Thursday 3/28 9am-5pm
  - Friday 3/29 9am-5pm

ETA: EOD Friday 3/29

**Comment by aqandrew** [comment]
# Analysis

`addEventListener`'s second parameter is typically a function ([MDN](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#syntax)). Because the `insertEventSchedule` function lacks a return statement, calling it inline in right-col-content.js with args in parentheses executes immediately but returns `undefined`.

# Recommendation

`update code`

Wrap the function call to `insertEventSchedule` with its arguments in an anonymous function:

```js
  document.addEventListener(
    "DOMContentLoaded",
    () => { insertEventSchedule(eventData, "events"); }
  );
```

## Testing

This update should be tested on every page that imports right-col-content.js:

- [right-col-content.js](https://github.com/hackforla/website/blob/07e652f184c743e5d934d7233a5b6772efc6e28c/assets/js/right-col-content.js)
  - imported in [right-col-content.html](https://github.com/hackforla/website/blob/23929c4b206d2ee1908f906070407c810e295d65/_includes/events-page/right-col-content.html#L32)
    - included in [events.html](https://github.com/hackforla/website/blob/a4add9bb669f94960dfa78a3b6c3c647878ac1ca/pages/events.html#L25)

- [x] Only the events page (`/events`) needs to be tested.
